background.ts

```diff
@@ -747,19 +771,248 @@ function newK8sManager() {
+    let changeNeeded = false;
+
+    for (const k in newSettings) {
+      const fqname = prefix ? `${ prefix }.${ k }` : k;
+
+      if (!(k in allowedSettings)) {
+        errors.push(`Setting name ${ fqname } isn't recognized.`);
+      } else if (typeof (allowedSettings[k]) === 'object') {
+        if (typeof (newSettings[k]) === 'object') {
+          changeNeeded = this.verifyProposedSettings(fqname, allowedSettings[k], newSettings[k], errors) || changeNeeded;
+        } else {
+          errors.push(`Setting ${ fqname } should wrap an inner object, but got <${ newSettings[k] }>.`);
+        }
+      } else if (typeof (newSettings[k]) === 'object') {
+        if (allowedSettings[k] === this.verifyObjectUnchanged) {
+          // Special case for things like `.WSLIntegrations` which have unknown fields.
+          changeNeeded = this.verifyObjectUnchanged(fqname, newSettings[k], errors) || changeNeeded;
+        } else {
+          // newSettings[k] should be valid JSON because it came from `JSON.parse(incoming-payload)`.
+          // It's an internal error (HTTP Status 500) if it isn't.
+          errors.push(`Setting ${ fqname } should be a simple value, but got <${ JSON.stringify(newSettings[k]) }>.`);
+        }
+      } else {
+        // Throw an exception if this field isn't a function, because in the verifier all values should be
+        // either child objects or functions.
+        changeNeeded = allowedSettings[k].call(this, fqname, newSettings[k], errors) || changeNeeded;
+      }
+    }
+
+    return changeNeeded;
+  }
+
+  protected verifyContainerEngine(fqname: string, desiredEngine: string, errors: string[]): boolean {
+    switch (desiredEngine) {
+    case 'containerd':
+    case 'moby':
+      break;
+    case 'docker':
+      desiredEngine = 'moby';
+      break;
+    default:
+      errors.push(`Invalid value for ${ fqname }: <${ desiredEngine }>; must be 'containerd', 'docker', or 'moby'`);
+
+      return false;
+    }
+
+    return cfg.kubernetes.containerEngine !== desiredEngine;
+  }
+
+  protected verifyEnabled(fqname: string, desiredState: string|boolean, errors: string[]): boolean {
+    if (typeof (desiredState) !== 'boolean') {
+      switch (desiredState.toLowerCase()) {
+      case 'true':
+        desiredState = true;
+        break;
+      case 'false':
+        desiredState = false;
+        break;
+      default:
+        errors.push(`Invalid value for ${ fqname }: <${ desiredState }>`);
+
+        return false;
+      }
+    }
+
+    return cfg.kubernetes.enabled !== desiredState;
+  }
+
+  protected verifyKubernetesVersion(fqname: string, desiredVersion: string, errors: string[]): boolean {
+    const ptn = /^v?(\d+\.\d+\.\d+)(?:\+k3s\d+)?$/;
+    let m = ptn.exec(desiredVersion);
+
+    if (!m) {
+      errors.push(`Desired kubernetes version not valid: <${ desiredVersion }>`);
+
+      return false;
+    }
+    desiredVersion = m[1];
+    m = ptn.exec(cfg.kubernetes.version);
```

Comment:
Not really, because the specified value might start with a 'v', and if it comes from `k3s-versions.json` it might contain a `k3s` suffix. Best to strip the `v` off the version in prefs, and compare the numerical part, which is all we care about.